src/JsonSchema/PointerResolver.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace JsonSchema;
+
+use JsonSchema\Exception\InvalidPointerException;
+use JsonSchema\Exception\ResourceNotFoundException;
+
+/**
+ * Resolve JSON Pointers (RFC 6901)
+ */
+class PointerResolver
+{
+    const EMPTY_ELEMENT = '_empty_';
+    const LAST_ELEMENT = '-';
+    const SEPARATOR = '/';
+
+    /**
+     * @var string
+     */
+    private $pointer;
+
+    /**
+     * Get the part of the document which the pointer points to.
+     *
+     * @param object $json    The json document to resolve within.
+     * @param string $pointer The Json Pointer.
+     *
+     * @throws InvalidPointerException
+     * @throws ResourceNotFoundException
+     *
+     * @return mixed
+     */
+    public function resolvePointer($json, $pointer)
+    {
+        if ($pointer === '') {
+            return $json;
+        }
+
+        $this->validatePointer($pointer);
+        $this->pointer = $pointer;
```

Comment:
I think it would be better to pass this pointer to `resolve()` instead of storing it. We don't really need to store this for any other reason but showing an exception message.